### PR TITLE
fix(editor): Prevent duplicate values in preview for SQL editor

### PR DESCRIPTION
**Pull request by elsmr**
## Summary

SQL query preview should not include duplicates (specifically brackets for this issue).

- Exclude bracket segments from preview
- Harden the `ExpressionEditorOutput` to not display duplicate segments

## Related tickets and issues
https://linear.app/n8n/issue/NODE-1286/postgres-sqleditor-duplicate-values-in-preview
https://community.n8n.io/t/postgresql-node-using-array-something-duplicates-the-output/43373

## Review / Merge checklist
- [ ] PR title and summary are descriptive. **Remember, the title automatically goes into the changelog. Use `(no-changelog)` otherwise.** ([conventions](https://github.com/n8n-io/n8n/blob/master/.github/pull_request_title_conventions.md))
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [ ] Tests included.
   > A bug is not considered fixed, unless a test is added to prevent it from happening again.
   > A feature is not complete without tests.

### CODE REVIEW

File: packages/editor-ui/src/components/InlineExpressionEditor/InlineExpressionEditorOutput.vue [hunk 1]
```diff
@@ -34,13 +34,18 @@ const i18n = useI18n();
 const editor = ref<EditorView | null>(null);
 const root = ref<HTMLElement | null>(null);
 
+function addSegment(str: string, segment: Segment): string {
+	const text = segment.kind === 'resolvable' ? String(segment.resolved) : segment.plaintext;
+	return str.substring(0, segment.from) + text + str.substring(segment.to + 1);
+}
+
```
Comment (ivov) [inline packages/editor-ui/src/components/InlineExpressionEditor/InlineExpressionEditorOutput.vue:41]: Found this: https://share.cleanshot.com/87tnHllq

Comment (ivov) [reply packages/editor-ui/src/components/InlineExpressionEditor/InlineExpressionEditorOutput.vue:41]: And this: https://share.cleanshot.com/FgVjXwWW

**Review by ivov — CHANGES_REQUESTED** [review]

File: packages/editor-ui/src/components/InlineExpressionEditor/InlineExpressionEditorOutput.vue (hunk 1 above)
Comment (elsmr) [reply packages/editor-ui/src/components/InlineExpressionEditor/InlineExpressionEditorOutput.vue]: Good finds! should've tested better 😓 fixed now!

**Review by netroy — COMMENTED** [review]
`resolvedExpression` in `packages/editor-ui/src/components/ExpressionEditorModal/ExpressionEditorModalOutput.vue` has the exact same code. Do we not need to update there as well?

**Comment by elsmr** [comment]
> `resolvedExpression` in `packages/editor-ui/src/components/ExpressionEditorModal/ExpressionEditorModalOutput.vue` has the exact same code. Do we not need to update there as well?

Good point, I extracted the common logic into a new component

**Review by ivov — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@1.39.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.39.0)

**Pull request merged**